src/main/java/com/snowflake/kafka/connector/SchematizationUtils.java

```diff
@@ -0,0 +1,176 @@
+package com.snowflake.kafka.connector;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import io.confluent.connect.avro.AvroConverterConfig;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.kafka.schemaregistry.avro.AvroSchemaProvider;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.avro.Schema;
+
+public class SchematizationUtils {
```

Comment:
If you get time, please add Javadocs to the public class.

Reply:
Thanks for refactoring

Reply:
I would add the Javadocs in the schema evolution PR